library/src/blas2/rocblas_ger_batched.hpp

```diff
@@ -0,0 +1,117 @@
+/* ************************************************************************
+ * Copyright 2016-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+#include "handle.h"
+#include "logging.h"
+#include "rocblas.h"
+#include "utility.h"
+
+template <typename T, typename U>
+__global__ void ger_kernel(rocblas_int m,
+                           rocblas_int n,
+                           U           alpha_device_host,
+                           const T* __restrict__ x,
+                           rocblas_int incx,
+                           const T* __restrict__ y,
+                           rocblas_int incy,
+                           T*          A,
+                           rocblas_int lda)
+{
+    ptrdiff_t tx = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    ptrdiff_t ty = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+
+    if(tx < m && ty < n)
+    {
+        auto alpha = load_scalar(alpha_device_host);
+        A[tx + lda * ty] += alpha * x[tx * incx] * y[ty * incy];
+    }
+}
+
+template <typename T, typename U>
+__global__ void ger_batched_kernel(rocblas_int m,
+                                   rocblas_int n,
+                                   U           alpha_device_host,
+                                   const T* const __restrict__ xa[],
+                                   rocblas_int incx,
+                                   const T* const __restrict__ ya[],
+                                   rocblas_int incy,
+                                   T* const    Aa[],
+                                   rocblas_int lda)
+{
+    ptrdiff_t tx = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    ptrdiff_t ty = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+
+    if(tx < m && ty < n)
+    {
+        auto alpha = load_scalar(alpha_device_host);
+        T*   A;
+        const T* __restrict__ x;
+        const T* __restrict__ y;
+        A = Aa[hipBlockIdx_z];
+        x = xa[hipBlockIdx_z];
+        y = ya[hipBlockIdx_z];
+

```

Comment:
It should be:

A = Aa[hipBlockIdx_z] + shiftA;
x = xa[hipBlockIdx_z] + shiftx;
y = ya[hipBlockIdx_z] + shifty;